### PR TITLE
feat: currency rates

### PR DESCRIPTION
**Pull request by Mansurkhoja**

### CODE REVIEW

File: src/store/modules/rate/index.js [hunk 1]
```diff
@@ -49,7 +50,10 @@ const actions = {
           .get(`${url}/getHistory?timestamp=${timestamp}`)
           .then((res) => {
             const rates = res?.data?.result[0]?.tickers ?? null
-            commit('setHistoryRates', { name: timestamp, value: rates })
+            if (timestamp !== undefined) {
```
Comment (adamant-al) [inline src/store/modules/rate/index.js:52]: Why are you doing a request if timestamp === undefined?
May be the first line of getHistoryRates — `if (!timestamp) return;`?

File: src/components/CurrencySwitcher.vue [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        @click="onSelect(idx)"
+      >
+        <v-list-tile-title>{{ currency }}</v-list-tile-title>
+      </v-list-tile>
+    </v-list>
+  </v-menu>
+</template>
+
+<script>
+// eslint-disable-next-line no-unused-vars
```
Comment (MerdyNumber1) [inline src/components/CurrencySwitcher.vue:42]: ?

File: src/filters/currencyAmountWithSymbol.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * Format currency amount to a fancy string with symbol
+ * @param {number} amount - Amount to format
+ * @param {string} symbol - Currency symbol (ticker)
+ * @param {boolean} isAdmBalance - Amount is ADM balance
+ * @returns {string}
+ */
+export default (amount, symbol = Cryptos.ADM, isAdmBalance) => {
+  if (amount !== undefined) {
+    return `${currencyAmount(amount, symbol, isAdmBalance)} ${symbol}`
+  } else return ''
```
Comment (MerdyNumber1) [inline src/filters/currencyAmountWithSymbol.js:14]: Please, use everywhere braces for if-else conditions

File: src/assets/stylus/themes/adamant/_inputs.styl [hunk 4]
```diff
@@ -17,6 +17,60 @@
...
+  margin-top: 4px
+  pointer-events: none
+  flex: 1 1 auto
+  font-size: 16px
+  text-align: left
+  display: flex
+  align-items: flex-start
+  min-height: inherit;
+  position: relative
+.fake-input__label
```
Comment (martiliones) [inline src/assets/stylus/themes/adamant/_inputs.styl:33]: Why don't you use child selectors everywhere? (`&__label`)

File: src/components/SendFundsForm.vue [hunk 5]
```diff
@@ -495,6 +502,27 @@ export default {
...
+        this.$store.commit('options/updateOption', {
+          key: 'currentRate',
+          value
+        })
+      }
+    },
+    transferFeeRate () {
+      const currentRate = this.$store.state.rate.rates[`${this.transferFeeCurrency}/${this.currentCurrency}`]
+      const feeRate = (this.transferFeeFixed * currentRate).toFixed(2)
+      return currentRate !== undefined ? `${feeRate} ${this.currentCurrency}` : ''
```
Comment (martiliones) [inline src/components/SendFundsForm.vue:520]: You can check if `currentRate` is undefined right after getting it from the store and immediately return the empty string. E.g.:

```js
const currentRate = this.$store.state.rate.rates[`${this.transferFeeCurrency}/${this.currentCurrency}`]

if (currentRate === undefined) {
  return ''
}

const feeRate = (this.transferFeeFixed * currentRate).toFixed(2)

return feeRate
```

File: src/components/SendFundsForm.vue [hunk 6]
```diff
@@ -495,6 +502,27 @@ export default {
...
+    },
+    finalAmountRate () {
+      const currentRate = this.$store.state.rate.rates[`${this.currency}/${this.currentCurrency}`]
+      const amountRate = (this.finalAmountFixed * currentRate).toFixed(2)
+      return currentRate !== undefined ? `${amountRate} ${this.currentCurrency}` : ''
```
Comment (martiliones) [inline src/components/SendFundsForm.vue:525]: Same as above

File: src/filters/currencyAmount.js [hunk 7]
```diff
@@ -13,6 +13,6 @@ export default (amount, symbol = Cryptos.ADM, isAdmBalance) => {
     const formatted = BigNumber(
       !isAdmBalance && symbol === Cryptos.ADM ? amount / 1e8 : amount
     ).toFixed()
-    return `${formatted} ${symbol}`
+    return `${formatted}`
```
Comment (martiliones) [inline src/filters/currencyAmount.js:16]: The `toFixed` method already returns a string. It should be `return formatted`

**Review by martiliones — CHANGES_REQUESTED** [review]

**Review by martiliones — APPROVED** [review]

**Review by adamant-al — APPROVED** [review]

**Pull request merged**